tests/src/utils.rs

```diff
@@ -0,0 +1,236 @@
+
+// target/debug/mirrord exec --pod-name pod_name  -c binary command
+pub fn start_node_server(pod_name: &str, command: Vec<&str>, env: HashMap<&str, &str>) -> Child {
+    let path = env!("CARGO_BIN_FILE_MIRRORD");
+    let args: Vec<&str> = vec!["exec", "--pod-name", pod_name, "-c"]
+        .into_iter()
+        .chain(command.into_iter())
+        .collect();
+    let server = Command::new(path)
+        .args(args)
+        .envs(&env)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+        .unwrap();
+    server
+}
+
+pub async fn setup_kube_client() -> Client {
+    let mut config = Config::infer().await.unwrap();
+    config.accept_invalid_certs = true;
+    let client = Client::try_from(config).unwrap();
+    client
+}
+
+// minikube service nginx --url
+pub async fn get_service_url(client: &Client, namespace: &str) -> Option<String> {
+    let service_api: Api<Service> = Api::namespaced(client.clone(), namespace);
+    let pod_api: Api<Pod> = Api::namespaced(client.clone(), namespace);
+    for pod in pod_api.list(&ListParams::default()).await.unwrap() {
+        if pod.metadata.name.unwrap().contains("nginx") {
+            let host_ip = pod.status.unwrap().host_ip.unwrap();
+            for service in service_api.list(&ListParams::default()).await.unwrap() {
+                if service.metadata.name.unwrap().contains("nginx") {
+                    let port = service.spec.unwrap().ports.unwrap()[0].node_port.unwrap();
+                    return Some(format!("http://{}:{}", host_ip, port));
+                }
+            }
+        }
+    }
+    None
+}
+
+// kubectl get pods | grep nginx
+pub async fn get_nginx_pod_name(client: &Client, namespace: &str) -> Option<String> {
+    let pod_api: Api<Pod> = Api::namespaced(client.clone(), namespace);
+    for pod in pod_api.list(&ListParams::default()).await.unwrap() {
+        if pod.metadata.name.as_ref().unwrap().contains("nginx") {
+            return Some(pod.metadata.name.unwrap());
+        }
+    }
+    None
+}
+
+// kubectl create namespace name
+pub async fn create_namespace(client: &Client, namespace: &str) {
+    let namespace_api: Api<Namespace> = Api::all(client.clone());
+    let new_namespace = Namespace {
+        metadata: ObjectMeta {
+            name: Some(namespace.to_string()),
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+    namespace_api
+        .create(&PostParams::default(), &new_namespace)
+        .await
+        .unwrap();
+}
+
+// kubectl delete namespace name
+pub async fn delete_namespace(client: &Client, namespace: &str) {
+    let namespace_api: Api<Namespace> = Api::all(client.clone());
+    namespace_api
+        .delete(namespace, &DeleteParams::default())
+        .await
+        .unwrap();
+}
+
+pub async fn reqwest_request(url: &str, method: Method) {
```

Comment:
A. try a clearer function name
B. as a rule, lets not have asserts in util functions, only in the tests themselves